src/main/java/io/github/fbiville/nestedrelationquerysdn6/ChangesetRepository.java

```diff
@@ -1,10 +1,13 @@
 package io.github.fbiville.nestedrelationquerysdn6;
 
-import org.springframework.data.repository.Repository;
+
+import org.springframework.data.neo4j.repository.Neo4jRepository;
 
 import java.util.Optional;
 
-public interface ChangesetRepository extends Repository<Changeset, Long> {
+public interface ChangesetRepository extends Neo4jRepository<Changeset, Long> {
```

Comment:
It's usually better to start with the bare minimum, i.e. a simple `Repository` and expose only the methods you need